hack/make/.integration-test-helpers

```diff
@@ -27,7 +27,35 @@ source "$MAKEDIR/.go-autogen"
 : ${TESTFLAGS:=}
 : ${TESTDEBUG:=}
 
-integration_api_dirs=${TEST_INTEGRATION_DIR:-"$(go list  -test -f '{{- if ne .ForTest "" -}}{{- .Dir -}}{{- end -}}'  ./integration/...)"}
+setup_integration_test_filter() {
+	if [ -z "${TEST_FILTER}" ]; then
+		return
+	fi
+
+	if [ -z "${TEST_SKIP_INTEGRATION}" ]; then
+		: ${TEST_INTEGRATION_DIR:=$(git grep "func\ ${TEST_FILTER}.*\(t\ \*testing\.T\)" ./integration | grep '_test\.go' | awk '{ print $1 }' | xargs -I file dirname ./file | uniq)}
+		if [ -z "${TEST_INTEGRATION_DIR}" ]; then
```

Comment:
Couldn't this be written somewhat simpler as something like:

```bash
if grep -rl "func\ ${TEST_FILTER}.*\(t\ \*testing\.T\)" ./integration | grep -q '_test\.go$'; then
    TESTFLAGS_INTEGRATION+="-test.run ${TEST_FILTER}"
else
    # skip ...
fi
```

(Since we only care if *something* matched, not necessarily what?)

IMO we should probably also avoid using `git grep` here since `.git` might not exist in places that are still useful for running this script (release archive tarballs, etc).

Reply:
I am interested in the directories for each because this populates TEST_INTEGRATION_DIR, but generally yes I like this!

Reply:
Ah right 👍

Reply:
Updated.